digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadwayDAO.scala

```diff
@@ -238,6 +238,32 @@ trait BaseRoadAddress {
 
     GeometryUtils.areAdjacent(nextStartPoint, currEndPoint, fi.liikennevirasto.viite.MaxDistanceForConnectedLinks)
   }
+
+  lazy val startingPoint: Point = (sideCode == SideCode.AgainstDigitizing, reversed) match {
+    case (true, true) | (false, false) =>
+      //reversed for both SideCodes
+      geometry.head
+    case (true, false) | (false, true) =>
+      //NOT reversed for both SideCodes
+      geometry.last
+  }
+  lazy val endPoint: Point = (sideCode == SideCode.AgainstDigitizing, reversed) match {
+    case (true, true) | (false, false) =>
+      //reversed for both SideCodes
+      geometry.last
+    case (true, false) | (false, true) =>
+      //NOT reversed for both SideCodes
+      geometry.head
+  }
+
+  def getEndPoints: (Point, Point) = {
```

Comment:
You can probably use the existing method getEndPoints for ProjectLink class

Reply:
No, I removed it from ProjectLinkDAO to this parent class BaseRoadAddress, because we need this getEndPoints for RoadAddress and ProjectLink classes